src/KeyringController.test.ts

```diff
@@ -65,56 +72,59 @@ describe('KeyringController', function () {
       expect(keyringController.keyrings).toHaveLength(0);
     });
 
-    it('emits "lock" event', async function () {
-      const spy = sinon.spy();
-      keyringController.on('lock', spy);
+    it('emits "lock" event', async () => {
+      const lockSpy = spy();
+      keyringController.on('lock', lockSpy);
 
       await keyringController.setLocked();
 
-      expect(spy.calledOnce).toBe(true);
+      expect(lockSpy.calledOnce).toBe(true);
     });
   });
 
-  describe('submitPassword', function () {
-    it('should not load keyrings when incorrect password', async function () {
+  describe('submitPassword', () => {
+    it('should not load keyrings when incorrect password', async () => {
       await keyringController.createNewVaultAndKeychain(PASSWORD);
       await keyringController.persistAllKeyrings();
       expect(keyringController.keyrings).toHaveLength(1);
 
       await keyringController.setLocked();
 
       await expect(
-        keyringController.submitPassword(`${PASSWORD}a`),
+        keyringController.submitPassword(`${PASSWORD as string}a`),
       ).rejects.toThrow('Incorrect password.');
       expect(keyringController.password).toBeUndefined();
       expect(keyringController.keyrings).toHaveLength(0);
     });
 
-    it('emits "unlock" event', async function () {
+    it('emits "unlock" event', async () => {
       await keyringController.setLocked();
 
-      const spy = sinon.spy();
-      keyringController.on('unlock', spy);
+      const unlockSpy = spy();
+      keyringController.on('unlock', unlockSpy);
 
       await keyringController.submitPassword(PASSWORD);
-      expect(spy.calledOnce).toBe(true);
+      expect(unlockSpy.calledOnce).toBe(true);
     });
   });
 
-  describe('persistAllKeyrings', function () {
-    it('should persist keyrings in _unsupportedKeyrings array', async function () {
-      const unsupportedKeyring = 'DUMMY_KEYRING';
-      keyringController._unsupportedKeyrings = [unsupportedKeyring];
+  describe('persistAllKeyrings', () => {
+    it('should persist keyrings in _unsupportedKeyrings array', async () => {
+      const unsupportedKeyring = { type: 'DUMMY_KEYRING', data: {} };
+      keyringController.unsupportedKeyrings = [unsupportedKeyring];
       await keyringController.persistAllKeyrings();
 
       const { vault } = keyringController.store.getState();
-      const keyrings = await mockEncryptor.decrypt(PASSWORD, vault);
-      expect(keyrings.indexOf(unsupportedKeyring) > -1).toBe(true);
+      const keyrings = (await mockEncryptor.decrypt(PASSWORD, vault)) as {
```

Comment:
Any way to avoid this typecast?

Reply:
[5b17f35](https://github.com/MetaMask/KeyringController/pull/202/commits/5b17f35678083e8e0cc27688f698e3fe5a9a4bdc)